### ISSUE TITLE
Failed validation: #/definitions/ChartResultObject

### ISSUE THREAD

**Issue by pazer001** [gadicc/node-yahoo-finance2#635]
# Validation Error

## Minimal Reproduction
await yahooFinance._chart('CGFFX', {
        period1: new Date('2022-07-03'),
        period2: new Date('2023-04-21'),
        interval: '1d',
      })

## Symbol(s) that it happened for
- CGFFX
- RGAEX
- RGEBX
- RGACX
- CGFEX

## Error Message
4|batch_analyze  | The following result did not validate with schema: #/definitions/ChartResultObject
4|batch_analyze  | [
4|batch_analyze  |   {
4|batch_analyze  |     "keyword": "yahooFinanceType",
4|batch_analyze  |     "message": "Expecting number'ish but got null",
4|batch_analyze  |     "params": {
4|batch_analyze  |       "schema": "number",
4|batch_analyze  |       "data": null
4|batch_analyze  |     },
4|batch_analyze  |     "instancePath": "/indicators/quote/0/high/200",
4|batch_analyze  |     "schemaPath": "#/definitions/ChartIndicatorQuote/properties/high/items/yahooFinanceType",
4|batch_analyze  |     "data": null
4|batch_analyze  |   },
4|batch_analyze  |   {
4|batch_analyze  |     "keyword": "yahooFinanceType",
4|batch_analyze  |     "message": "Expecting number'ish but got null",
4|batch_analyze  |     "params": {
4|batch_analyze  |       "schema": "number",
4|batch_analyze  |       "data": null
4|batch_analyze  |     },
4|batch_analyze  |     "instancePath": "/indicators/quote/0/low/200",
4|batch_analyze  |     "schemaPath": "#/definitions/ChartIndicatorQuote/properties/low/items/yahooFinanceType",
4|batch_analyze  |     "data": null
4|batch_analyze  |   },
4|batch_analyze  |   {
4|batch_analyze  |     "keyword": "yahooFinanceType",
4|batch_analyze  |     "message": "Expecting number'ish but got null",
4|batch_analyze  |     "params": {
4|batch_analyze  |       "schema": "number",
4|batch_analyze  |       "data": null
4|batch_analyze  |     },
4|batch_analyze  |     "instancePath": "/indicators/quote/0/open/200",
4|batch_analyze  |     "schemaPath": "#/definitions/ChartIndicatorQuote/properties/open/items/yahooFinanceType",
4|batch_analyze  |     "data": null
4|batch_analyze  |   },
4|batch_analyze  |   {
4|batch_analyze  |     "keyword": "yahooFinanceType",
4|batch_analyze  |     "message": "Expecting number'ish but got null",
4|batch_analyze  |     "params": {
4|batch_analyze  |       "schema": "number",
4|batch_analyze  |       "data": null
4|batch_analyze  |     },
4|batch_analyze  |     "instancePath": "/indicators/quote/0/close/200",
4|batch_analyze  |     "schemaPath": "#/definitions/ChartIndicatorQuote/properties/close/items/yahooFinanceType",
4|batch_analyze  |     "data": null
4|batch_analyze  |   },
4|batch_analyze  |   {
4|batch_analyze  |     "keyword": "yahooFinanceType",
4|batch_analyze  |     "message": "Expecting number'ish but got null",
4|batch_analyze  |     "params": {
4|batch_analyze  |       "schema": "number",
4|batch_analyze  |       "data": null
4|batch_analyze  |     },
4|batch_analyze  |     "instancePath": "/indicators/quote/0/volume/200",
4|batch_analyze  |     "schemaPath": "#/definitions/ChartIndicatorQuote/properties/volume/items/yahooFinanceType",
4|batch_analyze  |     "data": null
4|batch_analyze  |   },
4|batch_analyze  |   {
4|batch_analyze  |     "keyword": "yahooFinanceType",
4|batch_analyze  |     "message": "Expecting number'ish but got null",
4|batch_analyze  |     "params": {
4|batch_analyze  |       "schema": "number",
4|batch_analyze  |       "data": null
4|batch_analyze  |     },
4|batch_analyze  |     "instancePath": "/indicators/adjclose/0/adjclose/200",
4|batch_analyze  |     "schemaPath": "#/definitions/ChartIndicatorAdjclose/properties/adjclose/items/yahooFinanceType",
4|batch_analyze  |     "data": null
4|batch_analyze  |   }
4|batch_analyze  | ]
4|batch_analyze  | This may happen intermittently and you should catch errors appropriately.
4|batch_analyze  | However:  1) if this recently started happening on every request for a symbol
4|batch_analyze  | that used to work, Yahoo may have changed their API.  2) If this happens on
4|batch_analyze  | every request for a symbol you've never used before, but not for other
4|batch_analyze  | symbols, you've found an edge-case (OR, we may just be protecting you from
4|batch_analyze  | "bad" data sometimes stored for e.g. misspelt symbols on Yahoo's side).
4|batch_analyze  | Please see if anyone has reported this previously:
4|batch_analyze  |   https://github.com/gadicc/node-yahoo-finance2/issues?q=is%3Aissue+Failed%20validation%3A%20%23%2Fdefinitions%2FChartResultObject
4|batch_analyze  | or open a new issue (and mention the symbol):  yahoo-finance2 v2.3.10
4|batch_analyze  |   https://github.com/gadicc/node-yahoo-finance2/issues/new?labels=bug%2C+validation&template=validation.md&title=Failed%20validation%3A%20%23%2Fdefinitions%2FChartResultObject        
4|batch_analyze  | For information on how to turn off the above logging or skip these errors,
4|batch_analyze  | see https://github.com/gadicc/node-yahoo-finance2/tree/devel/docs/validation.md.

## Environment
Browser or Node: node
Node version (if applicable): 20.0.0
Npm version: 
Library version (e.g. `1.10.1`): latest

**Comment by gadicc** [comment]
Hey again @pazer001 and thanks for the report :)

Are you still experiencing this?
I wasn't able to reproduce with the same symbol, dates and interval.

Occasionally Yahoo's servers (temporarily) return errant data on some queries, it's one of the things validation can help catch.

Please let me know if you're still experiencing this and with what queries and I'll take another look.  Thanks!

**Comment by pazer001** [comment]
Hi @gadicc, i just reproduced it with "TA90.TA"

The following result did not validate with schema: #/definitions/ChartResultObject
[
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/high/113',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/high/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/high/118',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/high/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/high/128',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/high/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/high/158',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/high/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/high/163',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/high/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/high/168',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/high/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/high/242',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/high/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/low/113',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/low/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/low/118',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/low/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/low/128',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/low/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/low/158',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/low/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/low/163',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/low/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/low/168',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/low/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/low/242',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/low/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/open/113',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/open/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/open/118',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/open/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/open/128',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/open/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/open/158',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/open/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/open/163',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/open/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/open/168',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/open/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/open/242',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/open/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/close/113',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/close/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/close/118',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/close/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/close/128',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/close/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/close/158',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/close/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/close/163',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/close/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/close/168',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/close/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/close/242',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/close/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/volume/113',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/volume/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/volume/118',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/volume/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/volume/128',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/volume/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/volume/158',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/volume/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/volume/163',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/volume/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/volume/168',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/volume/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/quote/0/volume/242',
    schemaPath: '#/definitions/ChartIndicatorQuote/properties/volume/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/adjclose/0/adjclose/113',
    schemaPath: '#/definitions/ChartIndicatorAdjclose/properties/adjclose/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/adjclose/0/adjclose/118',
    schemaPath: '#/definitions/ChartIndicatorAdjclose/properties/adjclose/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/adjclose/0/adjclose/128',
    schemaPath: '#/definitions/ChartIndicatorAdjclose/properties/adjclose/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/adjclose/0/adjclose/158',
    schemaPath: '#/definitions/ChartIndicatorAdjclose/properties/adjclose/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/adjclose/0/adjclose/163',
    schemaPath: '#/definitions/ChartIndicatorAdjclose/properties/adjclose/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/adjclose/0/adjclose/168',
    schemaPath: '#/definitions/ChartIndicatorAdjclose/properties/adjclose/items/yahooFinanceType',
    data: null
  },
  {
    keyword: 'yahooFinanceType',
    message: "Expecting number'ish but got null",
    params: { schema: 'number', data: null },
    instancePath: '/indicators/adjclose/0/adjclose/242',
    schemaPath: '#/definitions/ChartIndicatorAdjclose/properties/adjclose/items/yahooFinanceType',
    data: null
  }
]

This may happen intermittently and you should catch errors appropriately.
However:  1) if this recently started happening on every request for a symbol
that used to work, Yahoo may have changed their API.  2) If this happens on
every request for a symbol you've never used before, but not for other
symbols, you've found an edge-case (OR, we may just be protecting you from
"bad" data sometimes stored for e.g. misspelt symbols on Yahoo's side).
Please see if anyone has reported this previously:

  https://github.com/gadicc/node-yahoo-finance2/issues?q=is%3Aissue+Failed%20validation%3A%20%23%2Fdefinitions%2FChartResultObject

or open a new issue (and mention the symbol):  yahoo-finance2 v2.3.10

  https://github.com/gadicc/node-yahoo-finance2/issues/new?labels=bug%2C+validation&template=validation.md&title=Failed%20validation%3A%20%23%2Fdefinitions%2FChartResultObject

For information on how to turn off the above logging or skip these errors,
see https://github.com/gadicc/node-yahoo-finance2/tree/devel/docs/validation.md.

**Comment by ob1wn** [comment]
I get this as well using v2.4.1 - noticed it when running '1m' intervals. I believe the issue is due to Yahoo showing a partial result candle in their charts, so until a candle has formed (e.g. the minute/hour/day past) it gives an interim result.

running `yahooFinance._chart('TSLA', {period1: '2023-06-27T18:00:00Z', interval: '1m'})`
I noticed that when looking at the last 2 timestamps, the one before last is the round minute (1687893120) while the other is 5 seconds later (1687893125) - yahoo timestamps a minute on the first second of said minute.

pulling the same data a minute later gives the same result, however if I set an end period (period 2) I get the correct results `yahooFinance._chart('TSLA', {period1: '2023-06-27T18:00:00Z', period2: '2023-06-27T19:22:00Z', interval: '1m'})`

**Comment by ob1wn** [comment]
Following up on my comment above, I found this work around...
```
const now = new Date();
const period2 = new Date(now.getTime() - (now.getTime() % 60000)).toISOString();
const chartData = await yahooFinance._chart('TSLA', {period1: '2023-06-27T13:30:00Z', period2, interval: '1m'});
```

**Comment by ob1wn** [comment]
If I end up using this lib I will push a PR, but from what I can tell, just need to change the schema for "ChartIndicatorQuote" setting the properties' "yahooFinanceType" to "number|null" fixed it locally for now see following
```
         "ChartIndicatorQuote": {
            "type": "object",
            "properties": {
                "high": {
                    "type": "array",
                    "items": {
                        "yahooFinanceType": "number|null"
                    }
                },
                "low": {
                    "type": "array",
                    "items": {
                        "yahooFinanceType": "number|null"
                    }
                },
                "open": {
                    "type": "array",
                    "items": {
                        "yahooFinanceType": "number|null"
                    }
                },
                "close": {
                    "type": "array",
                    "items": {
                        "yahooFinanceType": "number|null"
                    }
                },
                "volume": {
                    "type": "array",
                    "items": {
                        "yahooFinanceType": "number|null"
                    }
                }
            },
            "required": [
                "high",
                "low",
                "open",
                "close",
                "volume"
            ]
        },
```

**Comment by gadicc** [comment]
Thanks all for the reports, and especially @pazer001 for your patience and @ob1wn for your investigation here.

I'd like to get this fixed and ob1wn seems to have solved it, however, I'd really like to have a test for this and am still struggling to reproduce on my side.  If someone could send me a `symbol`, plus `period1` and `period2` values *as a microsecond epoch* (and `interval` if relevant), that produces this error, it will be greatly appreciated!

(Unfortunately for ob1wn's example, Yahoo don't retain 1m data from less than 30 days ago... that's my bad, apologies).

**Comment by z3nful** [comment]
This can be closed, was fixed in PR#662 by @spylacopa to use the code provided by @ob1wn above

**Comment by gadicc** [comment]
Ooh fantastic, thanks so much, @z3nful, that's really helpful! :raised_hands:
